### DISCUSSION TITLE
shortcut - search string not being replaced

### DISCUSSION THREAD

**Discussion by bshmng** [Ulauncher/Ulauncher#1367] [Q&A] [answered]
Hi,

Got the following shortcut defined:
`firefox -private-window  https://www.google.com/search?q=%s`

When typing a certain search string for example "abcd" it does show "https://www.google.com/search?q=abcd" in the ulauncher interface.
When running the shortcut though a private tab is being opened that searches for the search string "%s" instead of the applied search input. (Entering things directly in terminal resulting search is fine).

Am I overseeing something?

Best regards,
Bas

**Comment by friday** [comment, marked as answer]
In Ulauncher v5 `%s` is only supported for links, not scripts (Uluncher v6 supports them for both):

![image](https://github.com/Ulauncher/Ulauncher/assets/515120/ae9587c1-1b33-41ff-bf9d-c0cf82362271)

For scrips you should use a [shebang](https://linuxhandbook.com/shebang/), and your script will run though that, and depending on what it is you would use different ways to access the argument. If you use bash `$@` would work.

This should work for your use case:

```sh
#!/bin/bash
firefox -private-window https://www.google.com/search?q=$@
```
